ts/session/sending/PendingMessageCache.ts

```diff
@@ -1,36 +1,144 @@
+import { getItemById, createOrUpdateItem } from '../../../js/modules/data';
 import { RawMessage } from '../types/RawMessage';
-import { OutgoingContentMessage } from '../messages/outgoing';
+import { ContentMessage } from '../messages/outgoing';
+import * as MessageUtils from '../utils';
+import { PubKey } from '../types';
 
-// TODO: We should be able to import functions straight from the db here without going through the window object
+// This is an abstraction for storing pending messages.
+// Ideally we want to store pending messages in the database so that
+// on next launch we can re-send the pending messages, but we don't want
+// to constantly fetch pending messages from the database.
+// Thus we have an intermediary cache which will store pending messagesin
+// memory and sync its state with the database on modification (add or remove).
 
 export class PendingMessageCache {
-  private readonly cachedMessages: Array<RawMessage> = [];
+  private cache: Array<RawMessage>;
 
   constructor() {
-    // TODO: We should load pending messages from db here
+    // Load pending messages from the database
+    // You should await init() on making a new PendingMessageCache
+    //   if you'd like to have instant access to the cache
+    this.cache = [];
+    void this.init();
   }
 
-  public addPendingMessage(
-    device: string,
-    message: OutgoingContentMessage
-  ): RawMessage {
-    // TODO: Maybe have a util for converting OutgoingContentMessage to RawMessage?
-    // TODO: Raw message has uuid, how are we going to set that? maybe use a different identifier?
-    // One could be device + timestamp would make a unique identifier
-    // TODO: Return previous pending message if it exists
-    return {} as RawMessage;
+  public get(): Array<RawMessage> {
+    // Get all pending from cache, sorted with oldest first
+    return this.cache.sort((a, b) => a.timestamp - b.timestamp);
```

Comment:
```suggestion
    return [...this.cache].sort((a, b) => a.timestamp - b.timestamp);
```
also might be better to rename it to `getAllPending()`